source/sasl-auth.rst

```diff
@@ -0,0 +1,246 @@
+
+The following steps will help you configure your environment:
+
+.. contents::
+   :local:	  
+
+.. rubric:: Assumptions
+
+Before we move on to the configuration steps, we assume the following:
+
+1. You have the LDAP server up and running. The LDAP server is accessible to the server with |PSMDB| installed. This document focuses on OpenLDAP server. If you use Microsoft Windows Active Directory, refer to the :ref:`windows-ad` section for ``saslauthd`` configuration. 
+#. You must place these two servers behind a firewall as the communications between them will be in plain text. This is because the SASL mechanism of PLAIN can only be used when authenticating and credentials will be sent in plain text.
+#. You have ``sudo`` privilege to the server with the |PSMDB| installed.
+
+Configuring ``saslauthd``
+---------------------------------
+1. Install the SASL packages. Depending on your OS, use the following command:
+
+   For *RedHat and CentOS*:
+
+   .. code-block:: bash
+
+      $ sudo yum install -y cyrus-sasl cyrus-sasl-plain
+ 
+   For *Debian and Ubuntu*:
+
+   .. code-block:: bash
+
+      $ sudo apt-get install -y sasl2-bin
+		
+#. Configure SASL to use ``ldap`` as the  authentication mechanism.
+
+   .. note:: Back up the original configuration file before making changes.
+   
+   For *RedHat/CentOS*, specify the ``ldap`` value for the ``--MECH`` option using the following command:
+
+   .. code-block:: bash
+
+      $ sudo sed -i -e s/^MECH=pam/MECH=ldap/g /etc/sysconfig/saslauthd		
+
+   Alternatively, you can edit the :file:`/etc/sysconfig/saslauthd` configuration file:
+
+   .. code-block:: yaml
+
+      MECH=ldap		
+
+   For *Debian/Ubuntu*, use the following commands to enable the ``saslauthd`` to auto-run on startup and to set the ``ldap`` value for the ``--MECHANISMS`` option:
+
+   .. code-block:: bash
+
+      $ sudo sed -i -e s/^MECH=pam/MECH=ldap/g /etc/sysconfig/saslauthdsudo sed -i -e s/^MECHANISMS="pam"/MECHANISMS="ldap"/g /etc/default/saslauthdÂ 
+      $ sudo sed -i -e s/^START=no/START=yes/g /etc/default/saslauthd		
+
+   Alternatively, you can edit the :file:`/etc/default/sysconfig/saslauthd` configuration file:
+
+   .. code-block:: yaml
+
+      START=yes		
+      MECHANISMS="ldap"
+  
+#. Create the :file:`/etc/saslauthd.conf` configuration file and specify the settings that ``saslauthd`` requires to connect to a local LDAP service:
+
+OpenLDAP server
+**********************************
+
+The following is the example configuration file. Note that the server address **MUST** match the OpenLDAP installation:
+
+.. code-block:: text
+
+   ldap_servers: ldap://localhost
+   ldap_mech: PLAIN
+   ldap_search_base: dc=example,dc=com
+   ldap_filter: (cn=%u)
+   ldap_bind_dn: cn=admin,dc=example,dc=com
+   ldap_password: secret
+
+Note the LDAP password (``ldap_password``) and bind domain name (``ldap_bind_dn``).
+This allows the ``saslauthd`` service to connect to the LDAP service as root.
```

Comment:
it's not clear why these settings allow to connect as root, I'd remove this statement at all.

Reply:
@igorsol Please kindly assist here, shall we remove this statement or how can we rephrase it to explain what's going on here?

Reply:
Probably the root is mentioned because example `ldap_bind_dn` is admin? The next line clearly says "users should not store administartive passwords...".
I would rephrase this by replacing "as root" with "as administrator".